frontend/packages/core/src/AppNotifications/useCompareAppNotificationsData.tsx

```diff
@@ -0,0 +1,80 @@
+import get from "lodash/get";
+import isEmpty from "lodash/isEmpty";
+import isEqual from "lodash/isEqual";
+
+import { useUserPreferences } from "../Contexts/preferences-context";
+import type { AppBanners } from "../Types";
+
+const useCompareAppNotificationsData = (banners: AppBanners) => {
+  const { preferences, dispatch } = useUserPreferences();
+  const bannersPreferences: AppBanners = get(preferences, "banners");
+
+  const bannersData = {
+    header: {},
+    multiWorkflow: {},
+    perWorkflow: {},
+  };
+  let shouldUpdate = false;
+
+  if (!isEmpty(banners?.header)) {
+    const headerPreferences = {
```

Comment:
You think these could be handled with destructuring assignment?

Reply:
I decided not to use destructuring for this one because none of the attributes are mandatory so if the user forgets to add them on the json or place where this data is taken from then it will throw an error